app/src/main/java/org/simple/clinic/widgets/montyeardatepicker/MonthYearPickerDialog.kt

```diff
@@ -0,0 +1,298 @@
+package org.simple.clinic.widgets.montyeardatepicker
+
+import androidx.compose.foundation.background
+import androidx.compose.foundation.layout.Arrangement
+import androidx.compose.foundation.layout.Column
+import androidx.compose.foundation.layout.Row
+import androidx.compose.foundation.layout.Spacer
+import androidx.compose.foundation.layout.fillMaxWidth
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.layout.requiredHeight
+import androidx.compose.foundation.shape.RoundedCornerShape
+import androidx.compose.material.icons.Icons
+import androidx.compose.material.icons.filled.ArrowDropDown
+import androidx.compose.material3.AlertDialog
+import androidx.compose.material3.DropdownMenuItem
+import androidx.compose.material3.ExperimentalMaterial3Api
+import androidx.compose.material3.ExposedDropdownMenuBox
+import androidx.compose.material3.Icon
+import androidx.compose.material3.Text
+import androidx.compose.material3.TextButton
+import androidx.compose.material3.TextField
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.CompositionLocalProvider
+import androidx.compose.runtime.Stable
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.runtime.setValue
+import androidx.compose.runtime.staticCompositionLocalOf
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.draw.clip
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.platform.LocalTextInputService
+import androidx.compose.ui.res.colorResource
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.unit.dp
+import org.simple.clinic.R
+import org.simple.clinic.monthlyreports.form.compose.util.textFieldColors
+import org.simple.clinic.util.RealUserClock
+import org.simple.clinic.util.UserClock
+import org.simple.clinic.widgets.montyeardatepicker.MonthYearPickerValue.Month
+import org.simple.clinic.widgets.montyeardatepicker.MonthYearPickerValue.Year
+import java.time.LocalDate
+import java.time.YearMonth
+import java.time.ZoneId
+import java.time.format.DateTimeFormatter
+import java.time.format.TextStyle
+import java.util.Locale
+import java.time.Month as JavaMonth
+
+@OptIn(ExperimentalMaterial3Api::class)
+@Composable
+fun MonthYearPickerDialog(
+    onDismissRequest: () -> Unit,
+    modifier: Modifier = Modifier,
+    selectedDate: LocalDate? = null,
+    daysRange: IntRange = IntRange(0, 3650), // now to 10 years in future
+    onDateSelected: (LocalDate) -> Unit
+) {
+  val dateFormatter = remember {
+    DateTimeFormatter.ofPattern("MM/yyyy")
```

Comment:
Can we pass dateFormatter instance through constructor argument, this would help to initialise the instance through dagger (DateFormatterModule.kt)and make it consistent/similar to other parts of the app.

Reply:
That was intentional in this scenario. We don't want to use the Chronology in the date formatters. That's why went with a hardcoded one here. In future if we have a use case that needs support for both, we can revisit this.

Reply:
Think of it as a separate component similar to material date picker dialog. We just want to render it in this format and don't have anything else.